cypress/e2e/favouriting.cy.ts

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from 'uuid'
+
+describe('Model favouriting', () => {
+  const modelName = `Test model ${uuidv4()}`
+
+  before(() => {
+    cy.log('Navigate to Upload page and json tab')
+    cy.visit('/upload')
+    cy.get('[data-test=uploadJsonTab]').click({ force: true })
+
+    cy.log('Selecting schema and inputting metadata')
+    cy.get('[data-test=selectSchemaInput]').trigger('mousedown', { force: true, button: 0 })
+    cy.fixture('schema_names.json').then((schemaNames) => {
+      cy.get(`[role=option]:contains(${schemaNames.model})`).click()
+    })
+    cy.fixture('minimal_metadata.json').then((metadata) => {
+      const updatedMetadata = { ...metadata }
+      updatedMetadata.highLevelDetails.name = modelName
+      updatedMetadata.buildOptions.uploadType = 'Model card only'
+      cy.get('[data-test=metadataTextarea]')
+        .clear()
+        .type(JSON.stringify(metadata), { parseSpecialCharSequences: false, delay: 0 })
+    })
+
+    cy.log('Submitting model')
+    cy.get('[data-test=warningCheckbox]').click()
+    cy.get('[data-test=submitButton]').click()
+    cy.url({ timeout: 10000 }).as('modelUrl').should('contain', '/model/')
+  })
+
+  it('Is able to favourite and unfavourite a model', function favouriteModelTest() {
+    cy.visit(this.modelUrl)
```

Comment:
I think it's useful to check that the URL is as we expect it to be before continuing with the rest of the test as this acts like a "wait for the page to load" sort of thing. In my tests I've done the following:
```
  // import util function
  import convertNameToUrlFormat from '../utils/convertNameToUrlFormat'

  // then in your test you can do
  cy.visit(this.modelUrl)
    cy.fixture('minimal_metadata.json').then((modelMetadata) => {
      cy.url().should('contain', `/model/${convertNameToUrlFormat(modelMetadata.highLevelDetails.name)}`)
    })
```